modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/MultiSelectTag.java

```diff
@@ -0,0 +1,337 @@
+
+		String id = getId();
+
+		if (_label != null) {
+			jspWriter.write("<label");
+
+			if (Validator.isNotNull(id)) {
+				jspWriter.write(" for=\"");
+				jspWriter.write(id);
+				jspWriter.write("\"");
+			}
+
+			jspWriter.write(">");
+
+			jspWriter.write(_label);
+			jspWriter.write("</label>");
+		}
+
+		jspWriter.write("<div class=\"input-group\"><div class=\"");
+		jspWriter.write("input-group-item\"><div class=\"form-control ");
+		jspWriter.write("form-control-tag-group input-group\"><div class=\"");
+		jspWriter.write("input-group-item\">");
+
+		List<Map<String, String>> selectedItems = getSelectedItems();
+
+		if (!ListUtil.isEmpty(selectedItems)) {
+			for (Map<String, String> selectedItem : selectedItems) {
+				String selectedItemLabel = selectedItem.get("label");
+
+				LabelTag labelTag = new LabelTag();
+
+				labelTag.setDismissible(true);
+				labelTag.setLabel(selectedItemLabel);
+
+				labelTag.doTag(pageContext);
+
+				jspWriter.write("<input type=\"hidden\" name=\"");
+				jspWriter.write(selectedItemLabel);
+				jspWriter.write("\" ");
+				jspWriter.write("value=\"");
+				jspWriter.write(selectedItem.get("value"));
+				jspWriter.write("\" />");
+			}
+		}
+
+		jspWriter.write("<input class=\"form-control-inset\"");
+
+		if (_disabled) {
+			jspWriter.write(" disabled");
+		}
+
+		if (Validator.isNotNull(id)) {
+			super.writeIdAttribute();
+		}
+
+		jspWriter.write(" type=\"text\" /></div>");
+
+		if (!_disabled && !_disabledClearAll && !selectedItems.isEmpty()) {
+			jspWriter.write("<div class=\"input-group-item ");
+			jspWriter.write("input-group-item-shrink\">");
+
+			ButtonTag buttonTag = new ButtonTag();
+
+			buttonTag.setCssClass("component-action");
+			buttonTag.setDisplayType("");
+			buttonTag.setIcon("times-circle");
+
+			if (Validator.isNotNull(_clearAllTitle)) {
+				String clearAllTitle = LanguageUtil.get(
+					TagResourceBundleUtil.getResourceBundle(pageContext),
+					_clearAllTitle);
+
+				buttonTag.setTitle(HtmlUtil.escape(clearAllTitle));
+			}
+
+			buttonTag.doTag(pageContext);
+
+			jspWriter.write("</div>");
+		}
+
+		jspWriter.write("</div>");
+
+		if (_helpText != null) {
+			jspWriter.write("<div class=\"form-feedback-group\"><div ");
+			jspWriter.write("class=\"form-text\">");
+			jspWriter.write(_helpText);
+			jspWriter.write("</div></div>");
+		}
+
+		jspWriter.write("</div></div>");
+
+		return SKIP_BODY;
+	}
+
+	@Override
+	protected void writeDynamicAttributes() {
+	}
+
+	@Override
+	protected void writeIdAttribute() {
+	}
+
+	private static final String _ATTRIBUTE_NAMESPACE = "clay:multi-select:";
+
+	private String _clearAllTitle = "";
+	private String _closeButtonAriaLabel = "";
+	private boolean _disabled;
+	private boolean _disabledClearAll;
+	private String _helpText = "";
+	private String _inputName = "";
+	private String _inputValue = "";
+	private String _label = "";
+	private Object _locator;
+	private List<Map<String, String>> _selectedItems = new ArrayList<>();
```

Comment:
@carloslancha  Are these 2 classes: `MultiSelectItem` & `MultiSelectItemList` supposed to be used anywhere in the context of this PR? What is the purpose of these 2 classes?

Reply:
@kresimir-coko yes they are.

I guess we want 
```java
private MultiSelectItemList _selectedItems;
```
And each "item" in that list would have the type `MultiSelectItem`.
The purpose is to provide a type for these "items" and not just a `List` of `Map`, because a `Map` could contain "anything", does that make sense?

Reply:
Ah yea, makes sense, thanks @julien

Reply:
Hey @carloslancha I've fixed this in [4a0f0c3](https://github.com/liferay-frontend/liferay-portal/pull/528/commits/4a0f0c31c506f03a4a9738c7b166095af6d999ff).